app/scenes/messagesScene.js

```diff
@@ -19,21 +19,22 @@ export default class MessagesScene extends BaseScene<{}> {
     super(props);
     this.state = {
       messages: [],
-      showMessageDetailModal: false
+      showMessageDetailModal: false,
+      modalItem: []
     };
   }
 
   onMessageClick(item) {
     //Alert.alert('Item', JSON.stringify(item));
-    this.setState({ showMessageDetailModal: true });
+    this.setState({ showMessageDetailModal: true , modalItem: item});
 
     if(item.ReadDate == null){
       MessagesRequest.updateMessageReadDate(item.MessageId);
```

Comment:
Dodaj nakon ove linije da je ReadDate trenutno otvorene poruke različit od null, pošto nećemo dohvaćati sve poruke ponovno na zatvaranje modala, samo ćemo toj poruci promijeniti na taj način da je pročitana.

To možeš ovako:
```
      let messagesUpdated = this.state.messages;
      const index = messagesUpdated.findIndex(messageToUpdate => messageToUpdate.MessageId === item.MessageId);
      messagesUpdated[index].ReadDate = 1;
      this.setState({ messages: messagesUpdated});
```